src/charm.py

```diff
@@ -443,6 +458,11 @@ def request_certificates(self):
 
         bind_addrs = kubernetes_snaps.get_bind_addresses()
         common_name = kubernetes_snaps.get_public_address()
+        config_addrs = [
+            address
+            for option in ["loadbalancer-ips", "ha-cluster-vip", "ha-cluster-dns"]
+            for address in self.config[option].split()
+        ]
```

Comment:
is it reasonable to drop the address if they aren't configured?

```suggestion
            for address in self.config[option].split()
            if address
        ]
```

Reply:
Sure. The `if address` filter will just ensure that `address` is never something falsey like `None` or `""`. I don't think the return value from `.split()` should ever include those, but, I do like defensive programming so I'll go ahead and include this.

Reply:
too true.. it was a silly suggestion then :D